File: String/Remove_adjacent_duplicate_characters_from_a_string/removeDuplicateCharFromString.cpp
#include <iostream>
#include <string>
using namespace std;
 
// Function to remove adjacent duplicates characters from a string
void removeDuplicates(string &S) {
    char prev = '\0';
    for (auto it = S.begin(); it != S.end(); it++) {
        if (prev == *it) {
            S.erase(it);
            it--;
        }
        else {
            prev = *it;
        }
    }
}
 
int main() {
    string S = "AAABBCDDD";
    removeDuplicates(S);
    cout << S << endl;
 
    return 0;
}

File: String/Remove_adjacent_duplicate_characters_from_a_string/removeDuplicateCharFromString.py
# Function to remove adjacent duplicates char from a string
def removeDuplicates(S):
    chars = list(S)
    prev = None
    k = 0

    for c in S:
        if prev != c:
            chars[k] = c
            prev = c
            k = k + 1
    return "".join(chars[:k])


if __name__ == "__main__":
    S = "AAABBCCDDD"

    S = removeDuplicates(S)

    print(S)
